### DISCUSSION TITLE
The return value of uv_get_free_memory and uv_get_total_memory

### DISCUSSION THREAD

**Discussion by theanarkh** [libuv/libuv#3809] [Q&A]
The return value of `uv_get_free_memory` and `uv_get_total_memory` are `uint64_t`.  If the function fails it will return a negative number then it will be converted to `unsigned int64`. It will be confused with the normal value ?

**Comment by bnoordhuis** [comment]
They never fail, worst case they return zero when the answer is "don't know."

**Reply by theanarkh** [reply]
@bnoordhuis  But I find the function will fail in some platform, such as freebsd.
```
uint64_t uv_get_free_memory(void) {
  int freecount;
  size_t size = sizeof(freecount);

  if (sysctlbyname("vm.stats.vm.v_free_count", &freecount, &size, NULL, 0))
    return UV__ERR(errno);

  return (uint64_t) freecount * sysconf(_SC_PAGESIZE);

}
```

**Reply by bnoordhuis** [reply]
Oh, that's a bug. Want to send a PR?

**Reply by theanarkh** [reply]
> Oh, that's a bug. Want to send a PR?

Yes, I have submitted a PR. Thanks for reply.
